system.py

```diff
@@ -245,35 +253,37 @@ def BeamDetect(modeltype,goaltype,csvfile,dof,effdof,orderuse,E,rho,area,Im,reg=
 	# print(type(modedata),modedata.shape)
 	# print(eigvaluedata)
 	# print(type(kstiff),kstiff.shape)
-	numelem = 20     #number of elements
-	MeaNodes = [20]
-	effdof = [2*MeaNode-1  for MeaNode in MeaNodes]
-	numeig = 6
-
-	OrderUse = [i for i in range(numeig)]
-
-	eigenvalue_data= 2*math.pi*np.array([23.09, 140.90, 407.75, 795.14, 1292.30, 1998.43])
+	
+	effdof = [2*MeasuredNode-1  for MeasuredNode in MeasuredNodes]
+	orderuse = [i-1 for i in orderuse]
+	numeig = len(orderuse)
+
+	# mode_data = np.ones((1, numelem))
+	mode_data = data_[:,0].reshape((1,numeig))
+	# eigenvalue_data= 2*math.pi*np.array([23.09, 140.90, 407.75, 795.14, 1292.30, 1998.43])
+	eigenvalue_data= data_[:,1].reshape(numeig)
+	eigenvalue_data= eigenvalue_data*2*np.pi
 	eigenvalue_data= eigenvalue_data*eigenvalue_data
-	mode_data = np.ones((1, numelem))
-	eigvaluedata = eigenvalue_data[OrderUse]
-	modedata = mode_data[:, OrderUse]
-	weight = np.eye(numeig)
+	eigvaluedata = eigenvalue_data[orderuse]
+	modedata = mode_data[:, orderuse]
+	
 
-	Toll = 0.4953
-	Em=7.1e10
-	rhom=2.21e3
+	weight = np.eye(numeig)
+	
+	E=7.1e10
+	rho=2.21e3
+	area=0.0254*0.00635
 	Im=1/12*0.0254*0.00635**3
```

Comment:
The parameters `E`, `rho`, `area`, `Im` should be obtained from the input of function `BeamDetect()`.